test/test-core.js

```diff
@@ -247,6 +247,180 @@
 		});
 	});
 
+	t('Dynamic response status callback - list of statuses as an array (errors only)', function(assert) {
+		var done = assert.async();
+		var possibleStatuses = [500,400,404];
+	  	var returnedStatuses = [];
+		var maxNumLoops = possibleStatuses.length * 50;
+		var numLoops = 0;
+		var numLoopsComplete = 0;
+
+		$.mockjax({
+			url: '/response-callback',
+			response: function() {
+				this.status = possibleStatuses;
+				this.statusText = 'Internal Server Error';
+			}
+		});
+		
+		var completeCallback = function(xhr) {
+			assert.notEqual($.inArray(xhr.status, possibleStatuses), -1, 'Dynamically set random response status found');
+
+			if( $.fn.jquery !== '1.5.2') {
+				// This assertion fails in 1.5.2 due to this bug: http://bugs.jquery.com/ticket/9854
+				// The statusText is being modified internally by jQuery in 1.5.2
+				assert.equal(xhr.statusText, 'Internal Server Error', 'Dynamically set response statusText matches');
+			}
+
+			// add this to our array of returned statuses (if it isn't there already)
+			if($.inArray(xhr.status, returnedStatuses) === -1) {
+				returnedStatuses.push(xhr.status);
+			}
+
+			// increment counter
+			numLoopsComplete++;
+
+			// if we made it this far without matching all possible statuses, fail!
+			if (numLoopsComplete >= maxNumLoops) {
+				assert.equal(returnedStatuses.length, possibleStatuses.length, "Did not randomly return all possible statuses (only returned: " + returnedStatuses.toString() + ")");					
+				
+				done();
+			}
+		}
+
+		do {
+			$.ajax({
```

Comment:
Just a thought... but why not make these ajax calls synchronous? Most likely we'll only need to go through ~ n * 2 iterations, and with these being async we'll kick off a bunch of them even though we really only need n * 2 (or maybe *3 or *4). What do you think?